cips/cip-18.md

```diff
@@ -38,49 +38,62 @@ The following API is proposed for the standardised gas estimation service.
 
 ```proto
 service GasEstimator {
-    rpc EstimateGasUsage(EstimateGasUsageRequest) returns (EstimateGasUsageResponse) {}
-    rpc EstimateGasPrice(EstimateGasPriceRequest) returns (EstimateGasPriceResponse) {}
+    rpc EstimateFee(EstimateFeeRequest) returns (EstimateFeeResponse) {}
+    rpc EstimateFeeAndGas(EstimateFeeAndGasRequest) returns (EstimateFeeAndGasResponse) {}
 }
 
-message EstimateGasUsageRequest {
-    cosmos.tx.Tx tx = 1;
+enum TxPriority {
+  NONE = 0;
+  LOW = 1;
+  MEDIUM = 2;
+  HIGH = 3;
 }
 
-message EstimateGasUsageResponse {
-    uint64 estimated_gas_used = 1;
+message EstimateFeeRequest {
+    TxPriority tx_priority = 1;
 }
 
-message EstimateGasPriceRequest {}
+message EstimateFeeResponse {
+    double estimated_gas_price = 1;
+}
+
+message EstimateFeeAndGasRequest {
+    TxPriority tx_priority = 1;
+    cosmos.tx.Tx tx = 2;
```

Comment:
would this just run simulateTx?

Reply:
Yes

Reply:
given that the simulation just uses the bytes, why not just provide the tx bytes instead of the `sdk.Tx` type?

Reply:
You're right. I see it's been deprecated. Yeah let's change this to tx_bytes and the bytes format